test/images/sample-device-plugin/sampledeviceplugin.go

```diff
@@ -94,17 +94,26 @@ func main() {
 
 	}
 	dp1.SetAllocFunc(stubAllocFunc)
+	var registerControlFile string
 
-	if registerControlFile := os.Getenv("REGISTER_CONTROL_FILE"); registerControlFile != "" {
+	if registerControlFile = os.Getenv("REGISTER_CONTROL_FILE"); registerControlFile != "" {
 		if err := handleRegistrationProcess(registerControlFile); err != nil {
 			panic(err)
 		}
+		if err := dp1.Register(pluginapi.KubeletSocket, resourceName, pluginapi.DevicePluginPath); err != nil {
+			panic(err)
+		}
+		select {}
+	} else {
+		// if REGISTER_CONTROL_FILE environment variable is not explicitly specified, we would like to register the device
+		// plugin to kubelet as normal. The registration is added in a for loop to allow re-registration of the
+		// plugin to kubelet in case of node reboot or kubelet restarts.
+		for {
```

Comment:
do you need some sort of wait in-between of registrations? Otherwise it will be consuming a lot of CPU

Reply:
Sure, going to work on improvements.